content/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/chaingaurd-enforce-discovery-onboarding.md

```diff
@@ -0,0 +1,136 @@
+
+* `unsupported`: we do not currently support this type of cluster. 
+* `needs work`: we could support this cluster, but you need to take some steps before it is eligible. If any of your resources fall within this category, `chainctl` will outline what you need to change in order to make them eligible.
+* `eligible`: this cluster is ready to enroll with Chainguard Enforce.
+* `enrolled`: this cluster is already enrolled in Enforce.
+
+There are currently two options for how you can use Discovery: using the `chainctl cluster discover` command or the Chainguard Terraform provider.
+
+
+### Option 1 — `chainctl cluster discover`
+
+The `discover` subcommand requires you to specify which cloud provider you want to perform Discovery on. As of this writing, Chainguard Enforce Discovery supports the `gke`, `eks`, `ecs`, `cloudrun`, and `apprunner` options. In order to use Discovery, the GCP services require that you have set up GCP account associations. Likewise, the AWS services require that you have set up AWS impersonation.
+
+You can also include the `--group` option followed by the name of the IAM group under which you'd like to perform Discovery. Any clusters you enroll will be linked to this IAM group. If you don't specify a group, then you will be prompted to choose one.
+
+For example, you might run the following command to discover your GKE resources.
+
+```sh
+chainctl cluster discover --provider gke
+```
+```
+  PROVIDER |   	ACCOUNT    	|  LOCATION  | 	NAME  	| ELIGIBILITY | DETAILS  
+-----------+----------------------+------------+---------------+-------------+----------
+  	GKE | discovery-edu-375902 | US_CENTRAL | gcp-cluster-1 |	eligible |     	 
+   	GKE | discovery-edu-375902 | US_CENTRAL | gcp-cluster-2 |	eligible |     	 
+
+Would you like to enroll 2 eligible clusters?
+
+Do you want to continue? [Y,n]:
+```
+
+As this example shows, `chainctl` will prompt you to confirm whether you want to enroll all of the eligible resources. If you press `ENTER`, then all of the eligible clusters will be enrolled with Chainguard Enforce. 
+
+Be aware that the `discover` subcommand doesn’t currently have a good cleanup mechanism, meaning it doesn't provide a reliable method for deleting non-Kubernetes clusters.
+
+
+### Option 2 — Chainguard Terraform Provider
+
+The chainguard Terraform provider exposes a new “data source” to drive Chainguard Discovery. You can have Terraform read and use this data by adding the following block to your configuration. 
+
+```
+data "chainguard_cluster_discovery" "gotta-catch-em-all" {
+  id = local.group_id
+  providers = [
+	"APP_RUNNER",
+	"CLOUD_RUN",
+	"ECS",
+	"EKS",
+	"GKE",
+  ]
+  profiles = ["enforcer"]
+}
+```
+
+This will return the Discovery results, which can be piped into other resources (possibly as an output). The main use case for this data, though, will be to feed it into a “chainguard_cluster” resource:
```

Comment:
```suggestion
This will return the Discovery results, which can be piped into other resources (possibly as an output). The main use case for this data, though, will be to feed it into a `chainguard_cluster` resource:
```